revocation/crl/fetcher.go

```diff
@@ -44,6 +48,15 @@ type Fetcher interface {
 	Fetch(ctx context.Context, url string) (*Bundle, error)
 }
 
+// FetcherWithCertificateExtensions is an interface that specifies methods used
+// for fetching CRL from the given URL with certificate extensions to identify
+// the Freshest CRL extension (Delta CRL) from certificate.
+type FetcherWithCertificateExtensions interface {
```

Comment:
There 2 sources of Delta CRL URL:
1. certificate extension
2. CRL extension

This interface enable user to pass a certificate extension to the fetcher for accessing the delta CRL.